src/Microsoft.DotNet.Cli.Utils/DotnetVersionFile.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class DotnetVersionFile
+    {
+        public bool Exists { get; set; }
+
+        public string CommitSha { get; set; }
+
+        public string BuildNumber { get; set; }
+
+        /// <summary>
+        /// The runtime identifier (rid) that this CLI was built for.
+        /// </summary>
+        /// <remarks>
+        /// This is different than RuntimeEnvironment.GetRuntimeIdentifier() because the 
+        /// BuildRid is a RID that is guaranteed to exist and works on the current machine. The
+        /// RuntimeEnvironment.GetRuntimeIdentifier() may be for a new version of the OS that 
+        /// doesn't have full support yet.
+        /// </remarks>
+        public string BuildRid { get; set; }
+
+        public DotnetVersionFile(string versionFilePath)
+        {
+            Exists = File.Exists(versionFilePath);
+
+            if (Exists)
+            {
+                IEnumerable<string> lines = File.ReadLines(versionFilePath);
+
+                int index = 0;
+                foreach (string line in lines)
+                {
+                    if (index == 0)
+                    {
+                        CommitSha = line.Substring(0, 10);
+                    }
+                    else if (index == 1)
+                    {
+                        BuildNumber = line;
+                    }
+                    else if (index == 2)
+                    {
+                        BuildRid = line;
```

Comment:
The next iteration (if there is one) will break the loop.  The idea here is when we add another entry to this file, we will just need to add a "else if" before the else, parse the line, and everything will just work.  If we add a "break;" after reading the BuildRid, then the next person will need to remove it, in order to add a new entry.

The file gets written out by the build.  Currently there are 3 lines in the file:

https://github.com/dotnet/cli/pull/4281/files/283bf7139ed8e6432d4caf976fb8c513af5c9d36#diff-bbf7164ce4cfac6d95b9922845dcc746R266

```
var content = $@"{c.BuildContext["CommitHash"]}{Environment.NewLine}{version}{Environment.NewLine}{buildRid}{Environment.NewLine}";
File.WriteAllText(Path.Combine(sdkOutputDirectory, ".version"), content);
```